stdlib/public/core/Algorithm.swift

```diff
@@ -177,9 +177,8 @@ public struct EnumerateGenerator<
   ///
   /// - Requires: No preceding call to `self.next()` has returned `nil`.
   public mutating func next() -> Element? {
-    let b = base.next()
```

Comment:
@gribozavr Oops You are right. It will not return nil if our element is nil. Thanks for catching that.